src/layout/barGrid.js

```diff
@@ -87,16 +87,91 @@ export function prepareLayoutBarSeries(seriesType, ecModel) {
     return seriesModels;
 }
 
+
+/**
+ * Map from axis.index to min gap of two adjacent time values.
+ * For a single time axis, return value is in the form like
+ * [[1000000]].
+ * The value of 1000000 is in milliseconds.
+ */
+function getTimeAxesMinGaps(barSeries) {
+    /**
+     * Map from axis.index to time values.
+     * For a single time axis, axisValues is in the form like
+     * [[1495555200000, 1495641600000, 1495728000000]].
+     * Items in axisValues[x], e.g. 1495555200000, are time values of all
+     * series.
+     */
+    var axisValues = [];
+    zrUtil.each(barSeries, function (seriesModel) {
+        var cartesian = seriesModel.coordinateSystem;
+        var baseAxis = cartesian.getBaseAxis();
+        if (baseAxis.type !== 'time') {
+            return;
+        }
+
+        var data = seriesModel.getData();
+        var axisId = baseAxis.index;
+        for (var i = 0, cnt = data.count(); i < cnt; ++i) {
+            var value = data.get(baseAxis.dim, i);
+            if (!axisValues[axisId]) {
+                // No time value for the time axis
+                axisValues[axisId] = [value];
+            }
+            else if (axisValues[axisId].indexOf(value) < 0) {
+                // No time value in previous series
+                axisValues[axisId].push(value);
+            }
+            // Ignore duplicated time values in the same time axis
+        }
+    });
+
+    var axisMinGaps = [];
+    for (var i = 0; i < axisValues.length; ++i) {
+        if (axisValues[i]) {
+            // Sort time values into ascending order to calculate gaps
+            axisValues[i] = axisValues[i].sort(function (a, b) {
+                return a - b;
```

Comment:
A minor issue: js `sort` is in-pace. so you don't need to assign to `axisValues[i]` again.